apps/root/src/state/transactions/transactionUpdater.tsx

```diff
@@ -122,19 +133,86 @@ export default function Updater(): null {
     [transactions, dispatch, activeWallet?.address]
   );
 
+  const parseTxExtendedTypeData = useCallback(async (tx: TransactionDetails) => {
+    let extendedTypeData = {};
+
+    if (!tx.receipt) {
+      return extendedTypeData;
+    }
+
+    if (tx.type === TransactionTypes.newPair) {
+      extendedTypeData = {
+        id: toHex(tx.receipt.logs[tx.receipt.logs.length - 1].data),
+      };
+    }
+
+    if (tx.type === TransactionTypes.newPosition) {
+      const parsedLog = await transactionService.parseLog({
+        logs: tx.receipt.logs,
+        chainId: tx.chainId,
+        eventToSearch: 'Deposited',
+      });
+
+      if ('positionId' in parsedLog.args) {
+        extendedTypeData = {
+          id: parsedLog.args.positionId.toString(),
+        };
+      }
+    }
+
+    if (tx.type === TransactionTypes.migratePosition || tx.type === TransactionTypes.migratePositionYield) {
+      const parsedLog = await transactionService.parseLog({
+        logs: tx.receipt.logs,
+        chainId: tx.chainId,
+        eventToSearch: 'Deposited',
+      });
+
+      if ('positionId' in parsedLog.args) {
+        extendedTypeData = {
+          newId: parsedLog.args.positionId.toString(),
+        };
+      }
+    }
+
+    return extendedTypeData;
+  }, []);
+
   useEffect(() => {
     pendingTransactions.forEach((transaction) => {
       // eslint-disable-next-line @typescript-eslint/no-floating-promises
       positionService.setPendingTransaction(transaction);
     });
-    const nonPendingTransactions = Object.keys(transactions).filter((hash) => !!transactions[hash].receipt);
 
-    nonPendingTransactions.forEach((hash) => addTransactionToService(transactions[hash].receipt!, transactions[hash]));
-    dispatch(setInitialized());
+    nonPendingTransactions.forEach((tx) => addTransactionToService(tx.receipt!, tx));
 
+    dispatch(setInitialized());
     dispatch(setTransactionsChecking(pendingTransactions.map(({ hash, chainId }) => ({ hash, chainId }))));
   }, []);
 
+  useEffect(() => {
```

Comment:
I'm not sure this should go in here, since it doesnt have anything to do with "handlingTransactions" it could be a redux thunk action that just dispatched the state and we call it right after the fetch of the positions